processor/resourcedetectionprocessor/internal/aws/ec2/config.go

```diff
@@ -0,0 +1,22 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ec2
+
+// Config defines user-specified configurations unique to the EC2 detector
+type Config struct {
```

Comment:
Just for visibility for other reviewers: I want to have a flag to fetch all tags (right now it can be done with a `.*` filter but I think it's friendlier to have a separate flag for this). I will add this in a separate PR to keep the scope of this PR smaller.